defaults/main.yml

```diff
@@ -1 +1,11 @@
 ---
+
+# The yum repository and key from Icinga
+icinga2_agent_yum:
+  repo: "http://packages.icinga.com/epel/$releasever/release/"
+  key: "http://packages.icinga.com/icinga.key"
+
+# The apt repository and key from Icinga
+icinga2_agent_apt:
+  repo: "deb https://packages.icinga.com/{{ ansible_distribution|lower }} icinga-{{ ansible_distribution_release }} main"
```

Comment:
Usually https for apt doesn't work out of the box and requires additional packages to be installed first. I'd suggest to switch both to http and then overwrite it internally if required.